dist/spec/verifiable-payid-protocol-NC.txt

```diff
@@ -0,0 +1,896 @@
+              },
+            },
+            "signature": "base64Signature",
+          }
+          ]
+        "payload": {
+            "exp" : "1234574940",
+            "payId": "bob$wallet.com",
+            "payIdAddress": {
+              "expTime": "34874613475",
+              "paymentNetwork": "XRPL",
+              "environment": "TESTNET",
+              "addressDetailsType": "CryptoAddressDetails",
+              "addressDetails": {
+                "address": "T7CKYKhRujaxEs9fSxQwJApHsQVPKUgD7EtLWCGTAFBwTha"
+               }
+             }
+          }
+        }
+      ]
+  }
+
+
+
+
+
+Malhotra & Schwartz     Expires January 31, 2021               [Page 12]
+
+Internet-Draft     Verifiable PayID Protocol Extension         July 2020
+
+
+   In the above example, the "PaymentInformation" resource is a pre-
+   signed message with the PayID private keys of the PayID owner Bob.
+   Bob's non-custodial wallet retrieves this response and sends it to
+   the PayID client.
+
+5.  Security Considerations
+
+   This security considerations section only considers verifiable PayID
+   clients and servers bound to implementations as defined in this
+   document.
+
+   The security guarantees mentioned in [PAYID-PROTOCOL] applies to this
+   extension of verifiable PayID protocol.  In this section, we discuss
+   a security model for the Verifiable PayID protocol for non-custodial
+   service providers.
+
+5.1.  Security Model for Non-Custodial PayID Service Providers
+
+   In the current security model, non-custodial wallets do not store
+   their customers' keys on their servers.  The customers hold their
+   private keys on their device.  There is a no trust requirement
+   between the service provided by the non-custodial wallets and the
+   customers of this service.  Since the customers hold the private
+   keys: * The wallets are not liable for any consequences coming from
+   the lost, compromised or hacked private keys of the customers.  * The
+   non-custodial wallets do not require their customers to trust their
+   servers in case wallets servers go malicious or are compromised.
+
+   This extension of verifiable PayID protocol preserves this trust
+   model.  Consequently, a non-custodial wallet running a PayID server
+   has no liability for providing accurate "PaymentInformation" for
+   their customers that is signed with the private key of the non-
+   custodial PayID server wallet.  Instead, the PayID owners can
+   generate these signed mappings with their own PayID private key
+   locally on their app/device.  The sender of the payment (PayID client
+   wallet's customer) can easily verify these signatures based on the
+   trust relationship between the sender and the receiver (PayID owner).
+   This eliminates any risk of the non-custodial PayID server wallet
+   losing its private keys, going malicious, getting hacked, or becoming
+   otherwise compromised in a way that customers might lose funds.
```

Comment:
This protocol eliminating the risk of "Customers losing funds" feels a bit overstated to me. 

E.g., this protocol doesn't force a client to use any particular method to "trust an identity key", which could be a vector of losing funds, even if the protocol in this document is implemented perfectly.